leshan-client-core/src/main/java/org/eclipse/leshan/client/bootstrap/BaseBootstrapConsistencyChecker.java

```diff
@@ -37,24 +43,79 @@ public abstract class BaseBootstrapConsistencyChecker implements BootstrapConsis
 
     @Override
     public List<String> checkconfig(Map<Integer, LwM2mObjectEnabler> objectEnablers) {
+        List<String> errors = new ArrayList<>();
+
+        // validate if mandatory object enabler are present.
+        checkMandatoryObjectEnabler(objectEnablers, errors);
+
+        // validate device object
+        if (errors.isEmpty()) {
+            checkDeviceObjectEnabler(objectEnablers, errors);
+        }
+
+        // validate bootstrap config.
+        if (errors.isEmpty()) {
+            checkBootstrapConfig(objectEnablers, errors);
+        }
+
+        if (!errors.isEmpty()) {
+            return errors;
+        } else {
+            return null;
+        }
+    }
+
+    /**
+     * Check mandatory object enabler are present
+     */
+    protected void checkMandatoryObjectEnabler(Map<Integer, LwM2mObjectEnabler> objectEnablers, List<String> errors) {
```

Comment:
Why not check all objects tagged as mandatory in the model?

Reply:
fine :+1:  then yeah a comment would be welcome since the term "mandatory" could create confusion with the constraint from the object spec

Reply:
Done :)